github/commitstatus.go

```diff
@@ -57,8 +58,8 @@ type CommitStatus struct {
 //
 // See also:
 // https://docs.github.com/en/rest/commits/statuses#about-the-commit-statuses-api
-func NewCommitStatus(server, token, owner, repo, context string, log hclog.Logger) CommitStatus {
-	return CommitStatus{server, token, owner, repo, context, log}
+func NewCommitStatus(server, token, owner, repo, context string, log hclog.Logger, maxRetries int, waitTime, maxSleepTime time.Duration) CommitStatus {
```

Comment:
The number of arguments to the function is now too much.
Ideally, we should pass a HTTP Client instead, capable of retrying, but this would require a custom RoundTripper as I already mentioned.

A possibility would be to collect everything related to HTTP in a struct: server, maxRetries, waitTime, maxSleepTime, and call it maybe target, so that we would have
```
func NewCommitStatus(target target, token, owner, repo, context string, log hclog.Logger)
```

Reply:
Changed as suggested.